pkg/external-provider/provider.go

```diff
@@ -54,7 +55,14 @@ func (p *externalPrometheusProvider) GetExternalMetric(ctx context.Context, name
 	queryResults, err := p.promClient.Query(ctx, pmodel.Now(), selector)
 
 	if err != nil {
-		klog.Errorf("unable to fetch metrics from prometheus: %v", err)
+		if promErr, ok := err.(*prom.Error); ok { // Check if the error is of type *prom.Error
+			klog.Errorf("unable to fetch metrics from prometheus: %s", err.Error())
```

Comment:
ditto: Log all the info we have here, i.e. status code, query, error message, type